seqr/views/apis/igv_api.py

```diff
@@ -126,9 +130,37 @@ def fetch_igv_track(request, project_guid, igv_track_path):
     if igv_track_path.endswith('.bam.bai') and not does_file_exist(igv_track_path, user=request.user):
         igv_track_path = igv_track_path.replace('.bam.bai', '.bai')
 
+    if is_google_bucket_file_path(igv_track_path):
+        return _stream_gs(request, igv_track_path)
+
     return _stream_file(request, igv_track_path)
 
 
+def _stream_gs(request, gs_path):
+    headers = {'Authorization': 'Bearer {}'.format(_get_access_token(request.user))}
+    range_header = request.META.get('HTTP_RANGE')
+    if range_header:
+        headers['Range'] = range_header
+
+    bucket_object = gs_path.replace('gs://', '', 1).split('/', 1)
+    response = requests.get(
+        'https://{bucket}.storage.googleapis.com/{object}'.format(bucket=bucket_object[0], object=bucket_object[1]),
+        headers=headers,
+        stream=True)
+
+    return StreamingHttpResponse(response.iter_content(chunk_size=65536), status=response.status_code,
+                                 content_type='application/octet-stream')
+
+
+def _get_access_token(user):
+    if not hasattr(_get_access_token, 'timer') or (time.time() - _get_access_token.timer) > EXPIRATION_TIME_IN_SECONDS:
```

Comment:
I like this so much less than what you had before. If you disagree with my PR feedback, comment and explain why you don't want to go that route, do not just implement something totally different. This makes absolutely no sense in python - is this supposed to be a class? Are you making a function that also secretly can have properties assigned to it? This is incredibly un-pythonic - Functions should never have properties. Nothing should ever hav properties that may or may not exist the way these do

I suggested REDIS to handle caching and timeouts, because that is literally what redis does - it caches something for a fixed period of time, which is exactly what you are doing here. All you have done is reimplemented a cache, but like as a confusing set of object/ function properties. If you have a problem with redis, comment and explain why, don't go ahead and try to implement it on you own

Reply:
As I replied to your previous comment, the token will be used very frequently while IGV is loading a track. There are at least 20 separated loading requests for a gCNV track. Therefore, I don't like to add any extra time for each loading request. I even don't like to use Redis for the token because Redis using HTTP call to get the cached data which can have a significant delay.
If you like, I can change it to a class.

Reply:
please time redis and tell me what the percentage change would be. Quite frankly, going from 640ms to 660ms is a totally acceptable slow done

Reply:
Surprisingly to me, the difference is small. It just adds 5% more latency with Redis. Small packages increase much more delay while the large ones can't see any differences.
![image](https://user-images.githubusercontent.com/33550059/132059986-388a7b90-9339-456f-a2c4-61e883d3e508.png)

I will use Redis for the token.

Reply:
yeah, its good to know for future reference that redis is so quick!